### PR TITLE
Cleaned up various type conversion issues

### PR DESCRIPTION
**Pull request by slouken**
This makes sure SDL_PixelFormatEnum flows through the internal code correctly, as well as fixing a number of other minor issues.

### CODE REVIEW

**Comment by sezero** [comment]
Seeing those changes other than SDL_PixelFormatEnum, e.g. those casts: built the tree in c++ mode for this?

**Comment by sezero** [comment]
UWP fails: https://github.com/libsdl-org/SDL/actions/runs/8189254021/job/22393593680?pr=9218
```
D:\a\SDL\SDL\src\thread\windows\SDL_systhread.c(157,34): warning C4013: 'AddVectoredExceptionHandler' undefined; assuming extern returning int [D:\a\SDL\SDL\build\SDL\SDL3-static.vcxproj]
  (compiling source file '../../../src/thread/windows/SDL_systhread.c')
  
D:\a\SDL\SDL\src\thread\windows\SDL_systhread.c(157,32): warning C4047: '=': 'PVOID' differs in levels of indirection from 'int' [D:\a\SDL\SDL\build\SDL\SDL3-static.vcxproj]
  (compiling source file '../../../src/thread/windows/SDL_systhread.c')
  
D:\a\SDL\SDL\src\thread\windows\SDL_systhread.c(169,13): warning C4013: 'RemoveVectoredExceptionHandler' undefined; assuming extern returning int [D:\a\SDL\SDL\build\SDL\SDL3-static.vcxproj]
  (compiling source file '../../../src/thread/windows/SDL_systhread.c')

[...]

D:\a\SDL\SDL\src\video\winrt\SDL_winrtvideo.cpp(252,55): error C2440: '=': cannot convert from 'Uint32' to 'SDL_PixelFormatEnum' [D:\a\SDL\SDL\build\SDL\SDL3-static.vcxproj]
  (compiling source file '../../src/video/winrt/SDL_winrtvideo.cpp')
  D:\a\SDL\SDL\src\video\winrt\SDL_winrtvideo.cpp(252,55):
  Conversion to enumeration type requires an explicit cast (static_cast, C-style cast or parenthesized function-style cast)
  
D:\a\SDL\SDL\src\video\winrt\SDL_winrtvideo.cpp(301,23): error C2440: '=': cannot convert from 'DXGI_FORMAT' to 'SDL_PixelFormatEnum' [D:\a\SDL\SDL\build\SDL\SDL3-static.vcxproj]
  (compiling source file '../../src/video/winrt/SDL_winrtvideo.cpp')
  D:\a\SDL\SDL\src\video\winrt\SDL_winrtvideo.cpp(301,23):
  Conversion to enumeration type requires an explicit cast (static_cast, C-style cast or parenthesized function-style cast)
  
D:\a\SDL\SDL\src\video\winrt\SDL_winrtvideo.cpp(414,31): error C2440: '=': cannot convert from 'DXGI_FORMAT' to 'SDL_PixelFormatEnum' [D:\a\SDL\SDL\build\SDL\SDL3-static.vcxproj]
  (compiling source file '../../src/video/winrt/SDL_winrtvideo.cpp')
  D:\a\SDL\SDL\src\video\winrt\SDL_winrtvideo.cpp(414,31):
  Conversion to enumeration type requires an explicit cast (static_cast, C-style cast or parenthesized function-style cast)
```

**Comment by slouken** [comment]
> UWP fails: https://github.com/libsdl-org/SDL/actions/runs/8189254021/job/22393593680?pr=9218

The vectored exception handling warnings are fine, it's just because we have an out of date SDK on CI. The other issues are fixed, two of which were bugs that were caught by the switch to the enum.

**Comment by slouken** [comment]
> Seeing those changes other than SDL_PixelFormatEnum, e.g. those casts: built the tree in c++ mode for this?

Yep!

File: include/SDL3/SDL_events.h [hunk 1]
```diff
@@ -227,7 +227,11 @@ typedef enum
     /**
      *  This last event is only for bounding internal arrays
      */
-    SDL_EVENT_LAST    = 0xFFFF
+    SDL_EVENT_LAST    = 0xFFFF,
+
+    /* This just makes sure the enum is the size of Uint32 */
+    SDL_EVENT_ENUM_PADDING = 0x7FFFFFFF
```
Comment (sezero) [inline include/SDL3/SDL_events.h:233]: How did it pass `SDL_COMPILE_TIME_ASSERT(enum, sizeof(SDL_DUMMY_ENUM) == sizeof(int));` in SDL_stdinc.h before? Is this a platform with a 16 bit int?

Comment (slouken) [reply include/SDL3/SDL_events.h:233]: That assert doesn’t run on 3DS and Vita platforms. If we have an enumeration that needs to be a specific size, we need to do a little more work on it.

**Pull request merged**